src/littlefs/__main__.py

```diff
@@ -0,0 +1,134 @@
+import argparse
+import sys
+import textwrap
+import pathlib
+from littlefs import LittleFS
+
+
+def _add_default_args(parser: argparse.ArgumentParser):
+    parser.add_argument("-v", "--verbose", action="count", default=0)
+
+
+def _add_littlefs_args(parser: argparse.ArgumentParser):
+    parser.add_argument("-s", "--block-size", type=int, help="LittleFS block size")
+    parser.add_argument("-c", "--block-count", type=int, help="LittleFS block count")
+    parser.add_argument("-i", "--image", help="LittleFS filesystem image", type=pathlib.Path)

```

Comment:
Instead of `_add_default_args` and `_add_default_args` and adding them in a loop, you can define a common parser that is the parent of each subparser:

```
common_parser = argparse.ArgumentParser(add_help=False)
common_parser.add_argument("-v", "--verbose", action="count", default=0)
common_parser.add_argument("-s", "--block-size", type=int, help="LittleFS block size")
common_parser.add_argument("-c", "--block-count", type=int, help="LittleFS block count")
common_parser.add_argument("-i", "--image", help="LittleFS filesystem image", type=pathlib.Path)

parser_create = subparsers.add_parser(
    "create",
    parents=[common_parser,],
     help="Create LittleFS image from directory content",
)

# Do the same for the other commands
```

This, plus the other comments, it might be useful to define a helper function inside `main` that does this:

```
def add_command(handler):
     subparser = subparsers.add_parser(handler.__name__, parents=[global_parser])
     subparser.set_defaults(func=handler)
     return subparser
```

And then rename the command functions, e.g. `_create` -> `create`.

Reply:
And for `--block-size`, `--block-count`, and `--image`, I would set `required=True`.

Reply:
Thanks for this explanation! I search for something like this but did not find the correct way to do it, or was to tired to understand it...

Reply:
Would it be possible to find the block-size and block count automatically in `unpack` and `list`?

Reply:
it will be possible to infer block-count (but not block-size), once this is merged:

https://github.com/littlefs-project/littlefs/pull/866


I have a rough branch of littlefs-python that uses ^

https://github.com/BrianPugh/littlefs-python/tree/allow-optional-block-count

Once upstream littlefs merges my PR, i'll clean up that branch and make a PR to littlefs-python.